src/Microsoft.ML/LearningPipeline.cs

```diff
@@ -48,15 +49,26 @@ public ScorerPipelineStep(Var<IDataView> data, Var<ITransformModel> model)
     [DebuggerTypeProxy(typeof(LearningPipelineDebugProxy))]
     public class LearningPipeline : ICollection<ILearningPipelineItem>
     {
+        readonly internal IHostEnvironment Env;
```

Comment:
>Env [](start = 43, length = 3)

HostEnvironmentBase just happens to be IDisposable.
Since this design makes the pipeline own an instance of Env, your have to make the class disposable, 
Alternatively, you can take an optional environment and then the problem of owning env becomse the callers problem.  
If a user has to pass an environment to this class there are several potental benefits:
1. They can create custom, different environments and customize other things such as logging
2. That can share environments between different experiments
3.  If the pipeline is hosted within a macro, such as CV or Sweep, then the partent container of Pipeline can own the overall env.